pkg/sessions/dynamodb/dynamodb_store.go

```diff
@@ -0,0 +1,63 @@
+package dynamodb
+
+import (
+	"context"
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws/session"
+	dynamo "github.com/aws/aws-sdk-go/service/dynamodb"
+	"github.com/oauth2-proxy/oauth2-proxy/v7/pkg/apis/options"
+	"github.com/oauth2-proxy/oauth2-proxy/v7/pkg/apis/sessions"
+	"github.com/oauth2-proxy/oauth2-proxy/v7/pkg/sessions/persistence"
+	"time"
+)
+
+// SessionStore is an implementation of the persistence.Store
+// interface that stores sessions in redis
+type SessionStore struct {
+	Client Client
+}
+
+func NewDynamoDbSessionStore(opts *options.SessionOptions, cookieOpts *options.Cookie) (sessions.SessionStore, error) {
+	client, err := NewDynamoDbClient(opts.DynamoDb)
+	if err != nil {
+		return nil, fmt.Errorf("error constructing redis client: %v", err)
+	}
+
+	ss := &SessionStore{
+		Client: client,
+	}
+	return persistence.NewManager(ss, cookieOpts), nil
+}
+
+func NewDynamoDbClient(dynamodb options.DynamoDbStoreOptions) (Client, error) {
+
+	sess := session.Must(session.NewSessionWithOptions(session.Options{
+		SharedConfigState: session.SharedConfigEnable,
+	}))
+
+	client := dynamo.New(sess)
+
+	c := newClient(client)
+
+	return *c, nil
+}
+
+func (s *SessionStore) Save(ctx context.Context, s2 string, bytes []byte, duration time.Duration) error {
+	err := s.Client.Set(ctx, s2, bytes, duration)
+	if err != nil {
+		return fmt.Errorf("error saving dynamodb session: %v", err)
+	}
+	return nil
+}
+
+func (s *SessionStore) Load(ctx context.Context, s2 string) ([]byte, error) {
+	value, err := s.Client.Get(ctx, s2)
+	if err != nil {
+		return nil, fmt.Errorf("error loading dynamodb session: %v", err)
+	}
+	return value, nil
+}
+
+func (s SessionStore) Clear(ctx context.Context, s2 string) error {
```

Comment:
I know you have a placeholder in the client for `Del` too at the moment -- but this will likely just be boilerplate you can fill in now -- just copy over the sample from `redis_store.go` and update the text of the error.